# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1349979370 14400
#      Thu Oct 11 14:16:10 2012 -0400
# Node ID dca1f2bcccf30ab79966b65370e9bd46b1681138
# Parent  312d544de1650eb219bf25310d825f0ff4914133
don't register fltk toolkit if compiling without FLTK (bug #37541)

* __init_fltk__.cc (F__have_fltk__): New function.
Use it in PKG_ADD line to decide whether to register fltk toolkit.

diff --git a/libinterp/dldfcn/__init_fltk__.cc b/libinterp/dldfcn/__init_fltk__.cc
--- a/libinterp/dldfcn/__init_fltk__.cc
+++ b/libinterp/dldfcn/__init_fltk__.cc
@@ -24,17 +24,17 @@ along with Octave; see the file COPYING.
 
 To initialize:
 
   graphics_toolkit ("fltk");
   plot (randn (1e3, 1));
 
 */
 
-// PKG_ADD: register_graphics_toolkit ("fltk");
+// PKG_ADD: if (__have_fltk__ ()) register_graphics_toolkit ("fltk"); endif
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
 #include "defun-dld.h"
 #include "error.h"
 
@@ -2024,16 +2024,29 @@ DEFUN_DLD (__fltk_maxtime__, args, ,"")
         error ("argument must be a real scalar");
     }
 
   return retval;
 }
 
 #endif
 
+DEFUN_DLD (__have_fltk__, , , "")
+{
+  octave_value retval;
+
+#if defined (HAVE_FLTK)
+  retval = true;
+#else
+  retval = false;
+#endif
+
+  return retval;
+}
+
 // FIXME -- This function should be abstracted and made potentially
 // available to all graphics toolkits.  This suggests putting it in
 // graphics.cc as is done for drawnow() and having the master
 // mouse_wheel_zoom function call fltk_mouse_wheel_zoom.  The same
 // should be done for gui_mode and fltk_gui_mode.  For now (2011.01.30),
 // just changing function names and docstrings.
 
 DEFUN_DLD (mouse_wheel_zoom, args, ,
@@ -2121,9 +2134,8 @@ This function is currently implemented o
     error ("MODE must be one of the strings: \"2D\", \"3D\", or \"none\"");
 
   return octave_value (mode_str);
 #else
   error ("mouse_wheel_zoom: not available without OpenGL and FLTK libraries");
   return octave_value ();
 #endif
 }
-
